src/exportJarFileCommand.ts

```diff
@@ -75,139 +81,118 @@ export async function createJarFile(node?: INodeData) {
-        } catch (e) {
-            return reject(e);
-        }
-        const outputFilePath = join(outputPath, basename(outputPath) + ".jar");
-        await window.withProgress({
-            location: ProgressLocation.Window,
-            title: "Exporting Jar... : Generating jar...",
-            cancellable: true,
-        }, (progress, token) => {
-            return new Promise<string>(async (innerResolve, innerReject) => {
-                token.onCancellationRequested(() => {
-                    return reject();
-                });
-                const exportResult = await Jdtls.exportJar(basename(selectedMainMethod), outClassPaths, outputFilePath);
-                if (exportResult === true) {
-                    innerResolve(outputFilePath);
-                } else {
-                    innerReject("Export jar failed.");
-                }
+async function generateJar(pickSteps: string[], rootNodes: INodeData[],
+                           selectedMainMethod: string, outputPath: string): Promise<string | undefined> {
+    let outClassPaths: string[];
+    outClassPaths = await generateOutClassPath(pickSteps, rootNodes, outputPath);
+    const outputFilePath = join(outputPath, basename(outputPath) + ".jar");
+    await window.withProgress({
+        location: ProgressLocation.Window,
+        title: "Exporting Jar : Generating jar...",
+        cancellable: true,
+    }, (progress, token) => {
+        return new Promise<string>(async (resolve, reject) => {
+            token.onCancellationRequested(() => {
+                return reject();
             });
-        }).then((path) => {
-            resolve(path);
-        }, (err) => {
-            reject(err);
+            const exportResult = await Jdtls.exportJar(basename(selectedMainMethod), outClassPaths, outputFilePath);
+            if (exportResult === true) {
+                resolve(outputFilePath);
+            } else {
+                reject(new Error("Export jar failed."));
+            }
         });
     });
+    return outputFilePath;
 }
 
-function resolveMainMethod(pickSteps: string[], projectPath: string): Promise<string | undefined> {
-    return new Promise<string | undefined>(async (resolve, reject) => {
-        let mainMethods: MainMethodInfo[] = [];
-        await window.withProgress({
-            location: ProgressLocation.Window,
-            title: "Exporting Jar... : Resolving main classes...",
-            cancellable: true,
-        }, (progress, token) => {
-            return new Promise<string>(async (innerResolve, innerReject) => {
-                token.onCancellationRequested(() => {
-                    return innerReject();
-                });
-                mainMethods = await Jdtls.getMainMethod(projectPath);
-                innerResolve();
+async function resolveMainMethod(pickSteps: string[], projectPath: string): Promise<string | undefined> {
+    let mainMethods: MainMethodInfo[] = [];
+    await window.withProgress({
```

Comment:
Can we directly get the mainMethods from the invocation of `window.withProgress`?